server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -0,0 +1,363 @@
+package io.unitycatalog.server.persist;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.model.*;
+import io.unitycatalog.server.persist.dao.CatalogInfoDAO;
+import io.unitycatalog.server.persist.dao.RegisteredModelInfoDAO;
+import io.unitycatalog.server.persist.dao.SchemaInfoDAO;
+import io.unitycatalog.server.persist.utils.FileUtils;
+import io.unitycatalog.server.persist.utils.HibernateUtils;
+import io.unitycatalog.server.persist.utils.PagedListingHelper;
+import io.unitycatalog.server.persist.utils.RepositoryUtils;
+import io.unitycatalog.server.utils.ValidationUtils;
+import java.util.*;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ModelRepository {
+  private static final ModelRepository INSTANCE = new ModelRepository();
+  private static final Logger LOGGER = LoggerFactory.getLogger(ModelRepository.class);
+  private static final SessionFactory SESSION_FACTORY = HibernateUtils.getSessionFactory();
+  private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final PagedListingHelper<RegisteredModelInfoDAO> LISTING_HELPER =
+      new PagedListingHelper<>(RegisteredModelInfoDAO.class);
+
+  private ModelRepository() {}
+
+  public static ModelRepository getInstance() {
+    return INSTANCE;
+  }
+
+  public RegisteredModelInfo getRegisteredModelById(String registeredModelId) {
```

Comment:
Unused at the moment, but will be needed once we get model version support in place.